### ISSUE TITLE
Refactor FilteredConnection

### ISSUE THREAD

**Issue by lguychard** [sourcegraph/sourcegraph#9265]
The [`<FilteredConnection/>`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4927b5a0647a5c94410db477841f8cff363a52cf/-/blob/web/src/components/FilteredConnection.tsx) component is widely used in our codebase.

Unfortunately, it is currently very complex (almost 1K LOC!), untested, and contains several kludges / debtful parts:
- [updateOnChange](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4927b5a0647a5c94410db477841f8cff363a52cf/-/blob/web/src/components/FilteredConnection.tsx#L293-296)
- [Complex observable pipes with dual `scan()`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4927b5a0647a5c94410db477841f8cff363a52cf/-/blob/web/src/components/FilteredConnection.tsx#L503-618) to enable forward-direction cursor paging
- [Hacky calls to `setTimeout()`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4927b5a0647a5c94410db477841f8cff363a52cf/-/blob/web/src/components/FilteredConnection.tsx#L818-825)
- [Hacky computation of `totalCount`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4927b5a0647a5c94410db477841f8cff363a52cf/-/blob/web/src/components/FilteredConnection.tsx#L175-192)

We should refactor it to:
- Remove kludges
- Reduce complexity (eg. support forward pagination in a separate component)
- Be testable, and tested

**Comment by valerybugakov** [comment]
@umpox, should this issue be moved into the `Backlog (eng)` column on [our board](https://github.com/orgs/sourcegraph/projects/144#add-column) at least since we started making progress here?

**Comment by umpox** [comment]
Yep, good point. Will move it in - I'm going to split this into a tracking issue too
